src/runtime/runtime_wasi.go

```diff
@@ -0,0 +1,86 @@
+// +build wasm,wasi
+
+package runtime
+
+import "unsafe"
+
+type timeUnit float64
```

Comment:
This file is almost a duplicate of src/runtime/runtime_wasm.go. Can you separate out the bits that are different in js and in wasi in separate files? Something like:

runtime_wasm.go: contains all the code shared for WebAssembly targets (the vast majority)
runtime_wasi.go: contains just the WASI-specific bits
runtime_js.go: contains just the JS-specific bits